test/e2e/data/infrastructure-vsphere/main/ownerreferences/cluster-identity.yaml

```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereClusterIdentity
+metadata:
+  name: identity
```

Comment:
As discussed let's prefix, maybe let's just name it like the test: "ownerreferences" (same for secret, failuredomain, deploymentzone + if there are other objects which are not within our test namespace)